lib/fluent/plugin/buffer.rb

```diff
@@ -636,6 +637,25 @@ def clear_queue!
         end
       end
 
+      def evacuate_chunk(chunk)
+        # Overwrite this on demand.
+        #
+        # Note: Difference from the `backup` feature.
+        #       The `backup` feature is for unrecoverable errors, mainly for bad chunks.
+        #       On the other hand, this feature is for normal chunks.
+        #       The main motivation for this feature is to enable recovery by evacuating buffer files
+        #       when the retry limit is reached due to external factors such as network issues.
+        #
+        # Note: Difference from the `secondary` feature.
+        #       The `secondary` feature is not suitable for recovery.
+        #       It can be difficult to recover files made by `out_secondary_file` because the metadata
+        #       is lost.
+        #       For file buffers, the easiest way for recovery is to evacuate the chunk files as is.
+        #       Once the issue is recovered, we can put back the chunk files, and restart Fluentd to
+        #       load them.
+        #       This feature enables it.
+      end
```

Comment:
It would be better if this method has guard sentence, it would found the miss easily in future.

Like,
```ruby
  raise NotImplementedError, "BUG: buffer plugins MUST implement this method"
```

Reply:
Thanks!
However, it should be optional for buffer plugins to overwrite this method.

The default behavior of this method is to do nothing.
Buffer plugins can overwrite this on demand.

For example, `MemoryBuffer` does not overwrite this method.

If we force plugins to overwrite this method, it destroys the compatibility of existing third-party buffer plugins.